src/openms/source/ANALYSIS/ID/AScore.cpp

```diff
@@ -425,34 +471,47 @@ namespace OpenMS
     site_determining_ions[1].sortByPosition();
   }
 
-  Int AScore::numberOfMatchedIons_(const RichPeakSpectrum & th, const RichPeakSpectrum & windows, Size depth, double fmt)
+  Size AScore::numberOfMatchedIons(const RichPeakSpectrum & th, const RichPeakSpectrum & windows, Size depth, double fragment_mass_tolerance, bool fragment_mass_tolerance_ppm)
   {
-    Int n = 0;
+    Size n = 0;
     for (Size i = 0; i < windows.size() && i <= depth; ++i)
     {
       Size nearest_peak = th.findNearest(windows[i].getMZ());
-      if (nearest_peak < th.size() && fabs(th[nearest_peak].getMZ() - windows[i].getMZ()) < fmt)
-        ++n;
+      if (nearest_peak < th.size())
+      {
+        double theo_mz = th[nearest_peak].getMZ();
+        double error = abs(theo_mz - windows[i].getMZ());
+
+        if (fragment_mass_tolerance_ppm)
+        {
+          error = error / theo_mz * 1e6;
+        }
+
+        if (error < fragment_mass_tolerance)
+        {
+          ++n;
+        }
+      }
     }
     return n;
   }
 
-  double AScore::peptideScore_(std::vector<double> & scores)
+  double AScore::peptideScore(const std::vector<double> & scores) const
   {
```

Comment:
could you add an OPENMS_PRECONDITION here for the length to be more than 10? seems dangerous to me here :-)